packages/providers/src/transaction-response/transaction-response.ts

```diff
@@ -199,11 +192,40 @@ export class TransactionResponse {
     const result = await this.waitForResult<TTransactionType>(contractsAbiMap);
 
     if (result.isStatusFailure) {
+      // TODO: use FuelError
       throw new Error(
         `Transaction failed: ${(<FailureStatus>result.gqlTransaction.status).reason}`
       );
     }
 
     return result;
   }
+
+  async waitUntilTransactionProcessed(): Promise<void> {
+    while (this.gqlTransaction?.status?.type === 'SubmittedStatus') {
+      await this.sleepBasedOnAttempts();
+      await this.fetch();
+      this.attempts += 1;
+    }
+  }
+
+  async waitUntilResponseReceived(): Promise<void> {
+    while (!this.gqlTransaction?.status?.type) {
+      await this.sleepBasedOnAttempts();
+      await this.fetch();
+      this.attempts += 1;
+    }
+  }
```

Comment:
on this way attemps will advance twice as fast.

one solution is to have 2 different attempts like: `fetchAttempts` then pass it like `sleepBaseOnAttemps(this.fetchAttemps)`